File: src/de.rnd7.huemqtt/src/main/java/de/rnd7/huemqtt/mqtt/TemperatureMessage.java
package de.rnd7.huemqtt.mqtt;

import com.google.gson.annotations.SerializedName;

import java.math.BigDecimal;
import java.time.ZonedDateTime;

public class TemperatureMessage {

    public static TemperatureMessage fromState(final BigDecimal temperature, final ZonedDateTime lastUpdated) {
        final TemperatureMessage message = new TemperatureMessage();

        message.temperature = temperature;
        message.lastUpdated = lastUpdated;

        return message;
    }

    private BigDecimal temperature;

    @SerializedName("last-updated")
    private ZonedDateTime lastUpdated;

    public ZonedDateTime getLastUpdated() {
        return lastUpdated;
    }

    public BigDecimal getTemperature() {
        return temperature;
    }
}

File: lib/messages/temperature-message.ts
import { Temperature } from "../api/v2/types/temperature"

export type TemperatureMessage = {
    temperature: number
    "last-updated": string
}

export const fromTemperature = (temperature: Temperature) => {
    const message: TemperatureMessage = {
        temperature: Math.round(temperature.temperature.temperature * 100) / 100,
        "last-updated": new Date().toISOString()
    }

    return message
}
